server/api/api.go

```diff
@@ -2039,6 +2040,129 @@ func (a *API) handleUploadFile(w http.ResponseWriter, r *http.Request) {
 	auditRec.AddMeta("fileID", fileID)
 	auditRec.Success()
 }
+func (a *API) handleTeamBoardsInsights(w http.ResponseWriter, r *http.Request) {
+	// swagger:operation GET /teams/{teamID}/boards/insights getTeamUsers
+	//
+	// Returns team boards insights
+	//
+	// ---
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: teamID
+	//   in: path
+	//   description: Team ID
+	//   required: true
+	//   type: string
+	// - name: duration
+	//   in: query
+	//   description: duration of data to calculate insights for
+	//   required: true
+	//   type: string
+	// security:
+	// - BearerAuth: []
+	// responses:
+	//   '200':
+	//     description: success
+	//     schema:
+	//       type: array
+	//       items:
+	//         "$ref": "#/definitions/BoardInsight"
+	//   default:
+	//     description: internal error
+	//     schema:
+	//       "$ref": "#/definitions/ErrorResponse"
+
+	vars := mux.Vars(r)
+	teamID := vars["teamID"]
+	userID := getUserID(r)
+	query := r.URL.Query()
+	duration := query.Get("duration")
+
+	if !a.permissions.HasPermissionToTeam(userID, teamID, model.PermissionViewTeam) {
```

Comment:
I just want to confirm. If I can see the team, I can see the team insights?

Reply:
Guests won't have access to insights for now so @plant99 you can add another OR condition here also return this error response if the user is a guest.

Reply:
Added guests and license checks.